### PR TITLE
Use vocab term codec for URIs.

### PR DESCRIPTION
**Pull request by davidlehn**
Encode all URIs using the vocab term codec. This allows arbitrary URIs to be encoded with the same small integers as other terms. The mapping in a context needs to use a redundant form: `{"URI":"URI"}`. This technique assume a use case where the CBOR-LD size has high priority over context size.

I'm not sure if there are edge cases with the technique here.  (Need a better test suite.)  The URI codec is making assumptions when it sees a "value: int" structure that the int is a term id.  Would it ever be something else there?

I'm assuming this is breaking as well as the cryptosuite branch it's on top off.

### CODE REVIEW

File: lib/codecs/UriEncoder.js [hunk 1]
```diff
@@ -23,11 +24,17 @@ const SCHEME_TO_ENCODER = new Map([
 ]);
 
 export class UriEncoder extends CborldEncoder {
-  static createEncoder({value} = {}) {
+  static createEncoder({value, transformer} = {}) {
     if(typeof value !== 'string') {
       return false;
     }
 
+    // check vocab term map
```
Comment (dlongley) [inline lib/codecs/UriEncoder.js:32]:
```suggestion
    // first check vocab term map to maximize URL compression
```

**Review by dlongley — APPROVED** [review]
Thanks!
